glass/shells.py

```diff
@@ -626,14 +718,21 @@ def combine(
         Find weights for a given function.
 
     """
-    return sum(
-        np.expand_dims(weight, -1)
-        * np.interp(
-            z,  # type: ignore[arg-type]
-            shell.za,
-            shell.wa / np.trapz(shell.wa, shell.za),  # type: ignore[attr-defined]
-            left=0.0,
-            right=0.0,
-        )
-        for shell, weight in zip(shells, weights)  # type: ignore[arg-type]
+    return np.sum(  # type: ignore[no-any-return]
+        [
+            np.expand_dims(weight, -1)
+            * np.interp(
+                z,
+                shell.za,
+                shell.wa
+                / np.trapz(  # type: ignore[attr-defined]
+                    shell.wa,
+                    shell.za,
+                ),
+                left=0.0,
+                right=0.0,
+            )
+            for shell, weight in zip(shells, weights)
+        ],
+        axis=0,
```

Comment:
No need to use `np.sum()` and hold all of the intermediate arrays in memory at the same time.

Reply:
Not sure I follow? The problem is:

With `sum()` (not numpy one) the assumed output type is `int` so signature is wrong

The `[]` encapsulating the expression is because `mypy` errors with `Generator expression must be parenthesized`

Reply:
Ah, got it, this is changed for the implicit type